lib/src/layout_element.dart

```diff
@@ -28,52 +31,108 @@ class TableLayoutElement extends LayoutElement {
 
   @override
   Widget toWidget(RenderContext context) {
-    final colWidths = children
-        .where((c) => c.name == "colgroup")
-        .map((group) {
-          return group.children.where((c) => c.name == "col").map((c) {
-            final widthStr = c.attributes["width"] ?? "";
-            if (widthStr.endsWith("%")) {
-              final width =
-                  double.tryParse(widthStr.substring(0, widthStr.length - 1)) *
-                      0.01;
-              return FractionColumnWidth(width);
-            } else {
-              final width = double.tryParse(widthStr);
-              return width != null ? FixedColumnWidth(width) : null;
-            }
-          });
-        })
-        .expand((i) => i)
-        .toList()
-        .asMap();
+    final rows = <TableRowLayoutElement>[];
+    List<TrackSize> columnSizes;
+    for (var child in children) {
+      if (child is TableStyleElement) {
+        // Map <col> tags to predetermined column track sizes
+        columnSizes = child.children.where((c) => c.name == "col").map((c) {
+          final colWidth = c.attributes["width"];
+          if (colWidth != null && colWidth.endsWith("%")) {
+            final percentageSize =
+                double.tryParse(colWidth.substring(0, colWidth.length - 1));
+            return percentageSize != null
+                ? FlexibleTrackSize(percentageSize * 0.01)
+                : FlexibleTrackSize(1);
+          } else if (colWidth != null) {
+            final fixedPxSize = double.tryParse(colWidth);
+            return fixedPxSize != null
+                ? FixedTrackSize(fixedPxSize)
+                : FlexibleTrackSize(1);
+          } else {
+            return FlexibleTrackSize(1);
+          }
+        }).toList(growable: false);
+      } else if (child is TableSectionLayoutElement) {
+        rows.addAll(child.children.whereType());
+      } else if (child is TableRowLayoutElement) {
+        rows.add(child);
+      }
+    }
+
+    // All table rows have a height intrinsic to their (spanned) contents
+    final rowSizes =
+        List.generate(rows.length, (_) => IntrinsicContentTrackSize());
+
+    // Calculate column bounds
+    int columnMax = rows
+        .map((row) => row.children
+            .whereType<TableCellElement>()
+            .fold(0, (int value, child) => value + child.colspan))
+        .fold(0, max);
 
+    final cells = <GridPlacement>[];
+    final columnRowOffset = List.generate(columnMax + 1, (_) => 0);
+    int rowi = 0;
+    for (var row in rows) {
+      int columni = 0;
+      for (var child in row.children) {
+        if (columnRowOffset[columni] > 0) {
+          columnRowOffset[columni] = columnRowOffset[columni] - 1;
+          columni++;
+        }
+        if (child is TableCellElement) {
```

Comment:
I'm wondering what happens if child isn't a `TableCellElement`. I assume this means there are other tags that are within the table that are not table cells, and I'm wondering if an error would be appropriate in an `else` branch?

If my above assumption isn't correct, is the if just to appease the type system so you don't cast?

Reply:
Such element will be ignored. This is appropriate for empty text elements, for example. As for real ('misplaced') elements I am not sure if we should support that. A browser would place those outside the table layout, but that'd require modifying the dom I think.

Reply:
Ok, this sounds good fine to me :+1: